src/publicPlugin.ts

```diff
@@ -89,7 +89,7 @@ const publicPlugin: FastifyPluginAsync = async (fastify) => {
       // todo: use filter out of deleted items?
       const t2 = iTM.createGetManyTask(graaspActor);
       t2.getInput = () => ({
-        itemIds: t1.result.map(({ id }) => id),
+        itemIds: t1.result,
```

Comment:
A step you're missing is the task `GetItemIdsByCategoriesTask`. The db query indeed returns `{ id }[]` and the task "destructure" it, so that's way we can pass the return value of the task to `GetManyTask`!